File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int,int> m;
        int rem;
        for(int i=0;i<nums.size();i++){
            rem=target-nums[i];
            if(m.find(rem)!=m.end()){
                return {m[rem],i};
            }
            m[nums[i]]=i;
        }
        return {0,0};
    }
};

File: 1-two-sum/two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        hash={}
        for i in range(0,len(nums)):
            remaining=target-nums[i]
            if(remaining in hash):
                return [hash[remaining],i]
            hash[nums[i]]=i